modules/@ergo/core/components/Grid/Grid.vue

```diff
@@ -109,41 +152,198 @@ export default {
         },
+        columnsOffset() {
+            return this.isSelectColumn ? 1 : 0;
+        },
+        templateRows() {
+            const headerRowsTemplate = this.isBasicFilters ? `${ROW_HEIGHT.MEDIUM}px ${ROW_HEIGHT.MEDIUM}px` : `${ROW_HEIGHT.MEDIUM}px`;
+            const dataRowsTemplate = this.rowIds.length > 0 ? `repeat(${this.rowIds.length}, ${this.rowHeight}px)` : '';
+
+            return {
+                gridTemplateRows: `${headerRowsTemplate} ${dataRowsTemplate}`,
+            };
+        },
+        isColumnExists() {
+            const draggedElIndex = this.columns.findIndex(
+                column => column.id === this.draggedElement,
+            );
+
+            return draggedElIndex !== -1;
+        },
+    },
+    watch: {
+        isListElementDragging() {
+            if (this.isListElementDragging) {
+                this.addGhostColumn();
+            } else {
+                this.removeGhostColumn();
+            }
+        },
+    },
+    created() {
+        if (this.isSelectColumn) this.columnWidths.push(COLUMN_WIDTH.SELECT_ROW);
+
+        const { length } = this.columns;
+
+        for (let i = 0; i < length; i += 1) {
+            if (this.columns[i].id === GHOST_ID) {
+                this.columnWidths.push(COLUMN_WIDTH.GHOST);
+            } else {
+                this.columnWidths.push(COLUMN_WIDTH.DEFAULT);
+            }
+        }
+
+        if (this.isActionColumn) this.columnWidths.push(COLUMN_WIDTH.ACTION);
+    },
+    mounted() {
+        window.addEventListener('click', this.onClickOutside);
+    },
+    beforeDestroy() {
+        window.removeEventListener('click', this.onClickOutside);
     },
     methods: {
-        onInsertColumn(payload) {
-            this.$emit('insertColumn', payload);
+        ...mapActions('draggable', [
+            'setDraggableState',
+            'setDraggedElement',
+            'setDraggedElIndex',
+            'setGhostElXTranslation',
+            'setGhostIndex',
+        ]),
+        setEditingCellCoordinates(coordinates = { row: null, column: null }) {
+            this.editingCellCoordinates = coordinates;
+        },
+        getEditingCellCoordinates() {
+            return this.editingCellCoordinates;
+        },
+        onStickyChange({
+            isSticky, state,
+        }) {
+            if (state === PINNED_COLUMN_STATE.LEFT) {
+                this.isSelectColumnPinned = isSticky;
+            } else {
+                this.isActionColumnPinned = isSticky;
+            }
+        },
+        onClickOutside({ pageX, pageY }) {
+            const { gridColumns } = this.$refs;
+            const {
+                top, left, width, height,
+            } = gridColumns.getBoundingClientRect();
+
+            if (!(pageX > left
+                 && pageX < left + width
+                 && pageY > top
+                 && pageY < top + height)) {
+                this.setEditingCellCoordinates();
+            }
```

Comment:
we have ready method for this calculations : 
import { isMouseOutOfBoundsElement } from '@Core/models/drag_and_drop/helpers';